storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/serde/KafkaSqlKeySerializer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.storage.impl.ksql.serde;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+
+import org.apache.commons.io.output.UnsynchronizedByteArrayOutputStream;
+import org.apache.kafka.common.serialization.Serializer;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.apicurio.registry.storage.impl.ksql.keys.MessageKey;
+
+/**
+ * Responsible for serializing the message key to bytes.
+ * @author eric.wittmann@gmail.com
+ */
+public class KafkaSqlKeySerializer implements Serializer<MessageKey> {
+    
+    private static final ObjectMapper mapper = new ObjectMapper();
+    static {
+        mapper.setSerializationInclusion(Include.NON_NULL);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#serialize(java.lang.String, java.lang.Object)
+     */
+    @Override
+    public byte[] serialize(String topic, MessageKey messageKey) {
+        try {
+            UnsynchronizedByteArrayOutputStream out = new UnsynchronizedByteArrayOutputStream();
+            out.write(ByteBuffer.allocate(1).put((byte) messageKey.getType().ordinal()).array());
+            mapper.writeValue(out, messageKey);
```

Comment:
using json for the message key does not look like a good idea to me

Reply:
Agree.  See previous comment.  :)  I didn't want to boil the ocean with this PR.  I have a list of improvements that need to be made.  I will create GitHub issues for them soon.